app/pages/dashboard/PatientDrawer/useAgpCGM/buildGenerateAGPImagesFunction.js

```diff
@@ -0,0 +1,59 @@
+/* eslint-disable no-undef */
+
+import _ from 'lodash';
+import { utils as vizUtils } from '@tidepool/viz';
+import Plotly from 'plotly.js-basic-dist-min';
+import * as actions from '../../../../redux/actions';
+
+export const buildGenerateAGPImages = (dispatch) => {
+  const props = {
+    generateAGPImagesSuccess: (images) => dispatch(actions.sync.generateAGPImagesSuccess(images)),
+    generateAGPImagesFailure: (error) => dispatch(actions.sync.generateAGPImagesFailure(error)),
+  }
+
```

Comment:
This function is a rip of [this class method](https://github.com/tidepool-org/blip/blob/a95a74b3443f3f83e29491b779c0a4ea5edad8e5/app/pages/patientdata/patientdata.js#L922-L964) in PatientData.js.